### PR TITLE
fix: compilation on 32bit

### PR DESCRIPTION
**Pull request by wmillers**
Fix compilation on 32bit machine

### CODE REVIEW

**Comment by mangalaman93** [comment]
@wmillers Thanks for the PR. Two questions: 1. why are you using a 32 bit machine still? 2. Why does this create problems on a 32 bit machine?

**Comment by wmillers** [comment]
@mangalaman93

1. I need to run dgraph on a 32-bit system as part of my program.
2. When passing constants that exceed the range of int on a 32-bit system during compilation, the following error occurs:

```console
# github.com/dgraph-io/dgraph/xidmap
..\xidmap\trie.go:38:49: cannot use math.MaxUint32 (untyped int constant 4294967295) as int value in argument to z.NewBuffer(32 << 20, "Trie").WithMaxSize (overflows)
# github.com/dgraph-io/dgraph/worker
..\worker\restore_reduce.go:39:23: cannot use 2 << 30 (untyped int constant 2147483648) as int value in constant declaration (overflows)
..\worker\queue.go:305:42: cannot use math.MaxUint32 (untyped int constant 4294967295) as int value in argument to t.rng.Intn (overflows)
```

The `WithMaxSize` function (so as `Intn`) expects an argument of type int, and on a 32-bit system, the maximum allowed value is 2 << 30 - 1, which is math.MaxInt32.

**Comment by mangalaman93** [comment]
> @mangalaman93
> 
>     1. I need to run dgraph on a 32-bit system as part of my program.
> 
>     2. When passing constants that exceed the range of int on a 32-bit system during compilation, the following error occurs:
> 
> 
> ```
> # github.com/dgraph-io/dgraph/xidmap
> ..\xidmap\trie.go:38:49: cannot use math.MaxUint32 (untyped int constant 4294967295) as int value in argument to z.NewBuffer(32 << 20, "Trie").WithMaxSize (overflows)
> # github.com/dgraph-io/dgraph/worker
> ..\worker\restore_reduce.go:39:23: cannot use 2 << 30 (untyped int constant 2147483648) as int value in constant declaration (overflows)
> ..\worker\queue.go:305:42: cannot use math.MaxUint32 (untyped int constant 4294967295) as int value in argument to t.rng.Intn (overflows)
> ```
> 
> The `WithMaxSize` function (so as `Intn`) expects an argument of type int, and on a 32-bit system, the maximum allowed value is 2 << 30 - 1, which is math.MaxInt32.

Make sense. I am still wondering what machine you are planning to run that is still 32 bit. In any case, just one suggestion to this PR and we would be happy to merge it. Thanks for answering the questions.

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: worker/queue.go [hunk 1]
```diff
@@ -302,7 +302,7 @@ func (t *tasks) cleanup() {
 func (t *tasks) newId() uint64 {
 	myRaftId := State.WALstore.Uint(raftwal.RaftId)
 	for {
-		id := myRaftId<<32 | uint64(t.rng.Intn(math.MaxUint32))
+		id := myRaftId<<32 | uint64(t.rng.Intn(math.MaxInt32))
```
[the remark this answers is not in the document]
Comment (wmillers) [reply worker/queue.go]: Thanks for the suggestion, I have changed it to `Uint32()`.

**Review by mangalaman93 — APPROVED** [review]

**Review by sanjayk-github-dev — APPROVED** [review]
LGTM

**Pull request merged**